### PR TITLE
Add force_destroy to google_dns_managed_zone

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/1572

Any ideas on how to test this? The Record Sets needs to be created outside of terraform to test it effectively. Creating it via terraform is messy as you need to deal with order of execution etc.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dns: added `force_destroy` option to `google_dns_managed_zone` to delete records created outside of Terraform
```

### CODE REVIEW

File: templates/terraform/pre_delete/managed_dns_zone.go.erb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+			}
+
+			if rr.Type == "NS" {
+				mz, err := config.clientDns.ManagedZones.Get(project, zone).Do()
+				if err != nil {
+					return fmt.Errorf("Error retrieving managed zone %q from %q: %s", zone, project, err)
+				}
+				domain := mz.DnsName
+
+				if domain == rr.Name {
```
Comment (ndmckinley) [inline templates/terraform/pre_delete/managed_dns_zone.go.erb:30]: I think you can probably skip all NS records even if the DnsName doesn't match the ResourceRecordSet name.

**Comment by upodroid** [comment]
This was a fun change to make. 😁 It follows the same approach as the force_destroy testing on GCS Buckets.

The test is passing now.

```
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google-beta   master  24⬇  5✎  $    make testacc TEST=./google-beta TESTARGS='-run=TestAccDNSManagedZone_forceDestroy'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccDNSManagedZone_forceDestroy -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccDNSManagedZone_forceDestroy
--- PASS: TestAccDNSManagedZone_forceDestroy (13.17s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta 14.111s
```

File: templates/terraform/pre_delete/managed_dns_zone.go.erb (hunk 1 above)
Comment (upodroid) [reply templates/terraform/pre_delete/managed_dns_zone.go.erb:38]: The NS and SOA records can't be deleted.

**Comment by upodroid** [comment]
`make test` is returning a very long stack trace ~17000 lines

**Comment by ndmckinley** [comment]
Okay, I think that if you rebase this onto master, this should work.

**Comment by upodroid** [comment]
I rebased it

**Comment by ndmckinley** [comment]
Okay, I can see that that hasn't worked ... I'm not sure what the problem is.  The error message we're getting is `module requires go 1.14` - did you add a dependency with new version requirements somehow?

**Comment by ndmckinley** [comment]
Yeah, there are tons of changes in there that you didn't make - I think it's got something to do with the merges in the history of this branch.  Can you try squashing all your changes into a single commit - including all the merges - and force push that?

**Comment by ndmckinley** [comment]
Let me know when you have the git history sorted out and I'll rerun the test.

**Comment by upodroid** [comment]
@ndmckinley I think we are good to go. The bot is reporting the right downstream diffs.

**Comment by ndmckinley** [comment]
Okay - the history looks dodgy to me, still, but if the tests pass I guess it won't be an issue.

**Comment by ndmckinley** [comment]
Okay, most of the tests have already passed, I bet the rest also will, and are still running - I'll merge when they do.

**Comment by upodroid** [comment]
Yeah, it was tricky to get it right.

**Review by ndmckinley — APPROVED** [review]

**Pull request merged**